proto/v3/message_contents/invite.proto

```diff
@@ -0,0 +1,39 @@
+// V3 invite message structure
+syntax = "proto3";
+
+package xmtp.v3.message_contents;
+
+import "v3/message_contents/public_key.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/v3/message_contents";
+option java_package = "org.xmtp.proto.v3.message.contents";
+
+// InvitationV1 is the encrypted invitation message meant to be encrypted as
+// ciphertext in SealedInvitationV1 and decrypted by the recipient using the
+// provided inviter `InstallationContactBundle`
+message InvitationV1 {
+    // The inviter_wallet_address must match the address derived from the
+    // signature in the inviter contact bundle. Either the
+    // `inviter_wallet_address` or the `recipient_wallet_address` MUST match the
+    // wallet address of the account receiving this invite
+    string inviter_wallet_address = 1;
+    string recipient_wallet_address = 2;
+    string conversation_unique_id = 3;
+    // TODO: Decide whether we need a Context field
+}
+
+// InvitationEnvelopeV1 is the encrypted invitation message and the contact of
+// the sender
+message InvitationEnvelopeV1 {
+    // This contains the public key that will be used to decrypt the ciphertext
+    InstallationContactBundle inviter = 1;
```

Comment:
We only need the identity key to be able to decrypt the message, but having the whole contact bundle allows us to validate what wallet address invited you (since we can validate signatures). Without that, there's a lot of funny-business that can happen here.

Reply:
You're right that the `VmacAccountLinkedKey` is all we strictly need here.

It does feel tidier to have a single type to represent an installation's identity, rather than sometimes using a `VmacAccountLinkedKey` and sometimes using an `InstallationContactBundle`. If we break it up and only use a component of the contact bundle, we can't do things like cache the contact to save lookups later. There's more code paths to test around validation logic, etc.

Reply:
I guess what I'm getting at is that if you are the one initiating the conversation, the information you have about the other party is the `InstallationContactBundle` (which you need because you need the pre_key). But if you are on the receiving end, you would only have the `VmacAccountLinkedKey`. That's probably not the end of the world, but feels less clean.